### PR TITLE
RHOAIENG-58364: feat(kserve): apply LLMInferenceServiceConfig resources last during deploy

### PR DESCRIPTION
**Pull request by pierDipi**
## Description

During 3.3 → 3.4 upgrades, the kserve controller is updated first but the
old ValidatingWebhookConfiguration still points to kserve-webhook-server-service
which no longer serves the LLMInferenceServiceConfig validation endpoint.
Since WithApplyOrder places webhooks last, if LLMInferenceServiceConfig
resources are applied before the new ValidatingWebhookConfiguration replaces
the old one, validation fails and the operator stops — preventing the new
webhook configuration from ever being applied.

Add WithApplyOrderLLMInferenceServiceConfigLast which composes
SortByApplyOrder with a stable sort that pushes LLMInferenceServiceConfig
resources after webhooks. Add SortFn.Then to enable sort function composition.

https://redhat.atlassian.net/browse/RHOAIENG-58364

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has run the integration test pipeline and verified that it passed successfully
- [ ] New RELATED_IMAGE mappings are already listed in ODH-Build-Config and RHOAI-Build-Config, and links are included in PR description

### E2E test suite update requirement

When bringing new changes to the operator code, such changes are by default required to be accompanied by extending and/or updating the E2E test suite accordingly.

To opt-out of this requirement:
1. **Please inspect the [opt-out guidelines](https://github.com/opendatahub-io/opendatahub-operator/blob/main/docs/e2e-update-requirement-guidelines.md)**, to determine if the nature of the PR changes allows for skipping this requirement
2. If opt-out is applicable, provide justification in the dedicated `E2E update requirement opt-out justification` section below
3. Check the checkbox below:
- [x] Skip requirement to update E2E test suite for this PR
4. Submit/save these changes to the PR description. This will automatically trigger the check.

#### E2E update requirement opt-out justification

Existing E2E tests cover the change, unit tests added to verify ordering behavior.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved KServe deployment ordering to ensure inference service configurations are applied last in the deployment sequence, preventing potential configuration ordering issues.

* **Tests**
  * Added comprehensive test coverage for resource ordering functionality, including edge cases for empty and null inputs.

### CODE REVIEW

File: internal/controller/components/kserve/kserve_controller_actions_test.go [hunk 1]
```diff
@@ -708,6 +708,112 @@ func TestCheckOperatorAndCRDDependencies(t *testing.T) {
...
+		input := []unstructured.Unstructured{
+			newRes("apps", "v1", "Deployment", "deploy"),
+			newRes("", "v1", "Service", "svc"),
+			newRes("", "v1", "ConfigMap", "cm"),
+		}
+
+		result, err := sortLLMInferenceServiceConfigLast(ctx, input)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(result).To(HaveLen(3))
+	})
```
[the remark this answers is not in the document]
Comment (davidebianchi) [reply internal/controller/components/kserve/kserve_controller_actions_test.go:790]: Agree, we should check also the order. Something like this could work?


```suggestion
	t.Run("no LLMInferenceServiceConfig resources leaves order unchanged", func(t *testing.T) {
		g := NewWithT(t)
		ctx := t.Context()

		input := []unstructured.Unstructured{
			newRes("apps", "v1", "Deployment", "deploy"),
			newRes("", "v1", "Service", "svc"),
			newRes("", "v1", "ConfigMap", "cm"),
		}

		result, err := sortLLMInferenceServiceConfigLast(ctx, input)
		g.Expect(err).NotTo(HaveOccurred())
		g.Expect(result).To(Equal(input))
	})
```

**Review by davidebianchi — APPROVED** [review]
